pkg/apiserver/rest/dualwriter_mode3.go

```diff
@@ -60,3 +59,36 @@ func (d *DualWriterMode3) Get(ctx context.Context, name string, options *metav1.
 
 	return legacy.Get(ctx, name, &metav1.GetOptions{})
 }
+
+// Update overrides the generic behavior of the Storage and writes first to US and then to the legacy store.
+func (d *DualWriterMode3) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+	old, err := d.Storage.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, false, err
+	}
+
+	updated, err := objInfo.UpdatedObject(ctx, old)
+	if err != nil {
+		return nil, false, err
+	}
+	objInfo = &updateWrapper{
+		upstream: objInfo,
+		updated:  updated,
+	}
+
+	obj, created, err := d.Storage.Update(ctx, name, objInfo, createValidation, updateValidation, forceAllowCreate, options)
+	if err != nil {
+		klog.FromContext(ctx).Error(err, "could not write to US", "mode", Mode3)
+		return obj, created, err
+	}
+
+	legacy, ok := d.Legacy.(rest.Updater)
```

Comment:
I think it's worth checking this right at the start so that we don't proceed with updates to Storage unless we know LegacyStorage also implements `rest.Updater`. Otherwise if this check fails we are guaranteed to only have a partial update (Storage but not LegacyStorage).

Reply:
I also thought about that, but after that I thought that in mode3, we will want for the Unified Storage to be already the source of truth (since we will be reading from it already), so if we can't update the legacy one, it should not matter, no? I'd actually go further and even consider, instead of returning `legacy.Update()`, call the function and just log on error but not return an error... wdyt?

Reply:
We'll need to look at failure modes more closely as a future step, but for now we're thinking about Storage being best effort during mode 2 and LegacyStorage being best effort during mode 3. That means that in mode 3 we could in fact just log the LegacyStorage error and not return it (similar to what we do for Storage errors in mode 2).  

However, there's a difference between an error caused by a one-off failure of LegacyStorage versus LegacyStorage not implementing the Updater interface and therefore _all_ updates on the LegacyStorage side failing. The second case is what we would have by doing the type casting of LegacyStorage so late in the DW Update implementation. That's why I recommend doing the rest.Updater type casting of LegacyStorage as the first step. Wdyt?

Reply:
Realistically the simplest answer here may be to just require that legacystorage implement all methods so we can remove all the checks at this level. If there genuinely is a case where a given method isn't supported by legacy storage it can return an error.

I do agree that in mode 3 updating legacy storage is best-effort and a failure should be logged but not impact the end user.